app/helpers/application_helper.rb

```diff
@@ -433,7 +433,13 @@ def hidden_spinner(text = '', options = {})
   end
 
   def hosts_count(resource_name = controller.resource_name)
-    @hosts_count ||= Host::Managed.reorder('').authorized.group("#{resource_name}_id").count
+    # If we are on /organizations or /locations, this allows to display the
+    # count for hosts not in the current organization & location.
+    if ['organization', 'location'].include? resource_name
+      hosts_scope = Host::Managed.reorder('').unscoped
```

Comment:
Oh, actually I was wrong, there is another `.authorized` below. However there is some repetition here, how about 
```ruby
hosts_scope = Host::Managed.reorder('')
hosts_scope = hosts_scope.unscoped if ['organization', 'location'].include? resource_name
@hosts_count ||= hosts_scope.authorized.group("#{resource_name}_id").count
```